az-digital/az_quickstart
# Fixes #2524 Author Entity Autolabel Length

`az_author` entities have a hidden name field that is used administratively, which is built from tokens by `entity_autolabel` from the author's name components. There is an issue because the length of this name field is `50`, whereas `entity_autolabel` is hardcoded to use a length of `255` in its truncation, as this is the length node uses. This means a SQL exception is generated when `entity_autolabel` generates a very long author name. This PR updates the length of the author name field to be the same length as the node entity's title. 

This has a side effect of generally giving better support for extremely long names.

A database update is necessary because there may be existing data in the database while we are altering the schema.

## Description
This PR updates the length of the `az_author` entity `name` field to `255` and applies a database update that updates the stored field storage definitions, ,key-value store SQL storage, and then finally applies the actual schema change. This is necessary because `updateFieldStorageDefinition()` cannot operate on data that already exists.

## Related issues
#2524 

## How to test

- Begin with a site that **does not** include this PR. 
- Enable publications, create at least one publication with some authors.
- Apply this PR to the environment.
- Verify in the database that the size of `az_author_field_data` field `name` is 50 characters (e.g. `lando mysql`, `use drupal10`, `describe az_author_field_data`)
- Verify in the database that the size of `az_author_field_revision` field `name` is 50 characters
- Apply database updates via `lando drush updb`
- Verify that update `1020701` runs without error
- Verify in the database that the size of `az_author_field_data` field `name` is 255 characters
- Verify in the database that the size of `az_author_field_revision` field `name` is 255 characters
- Verify that the contents of the table still contain the names of the authors you created
- Verify that modifying your publication and its authors appears to work as normal.
- Visit `/admin/reports/status`
- Verify that the following message **does not** appear as a warning:
```
Mismatched entity and/or field definitions
The following changes were detected in the entity type and field definitions.
Author - The Name field needs to be updated.
```

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
